source-mongodb/pull.go

```diff
@@ -412,25 +482,25 @@ func (c *capture) BackfillCollection(ctx context.Context, client *mongo.Client,
 			if err = c.Output.Checkpoint(checkpointJson, true); err != nil {
```

Comment:
With backfills running concurrently, these calls to `Documents` and `Checkpoint` will be interleaved and one binding could potentially commit another one's documents etc. It's wouldn't be the end of the world since the result would be a possible duplication of documents, but still probably good to avoid. I think what we'll need to do is group up all of the returned documents from a single mongo network-batch send send them atomically with the corresponding checkpoint update.